src/components/ArticleInfo.jsx

```diff
@@ -0,0 +1,50 @@
+import React, { useState, useEffect, useContext } from "react";
+import { useParams, Link } from "react-router-dom";
+import { getArticleById } from "../api";
+import { Loading } from "./Loading";
+import { LoadingContext } from "../context/LoadingProvider";
+
+export const ArticleInfo = () => {
+  const { article_id } = useParams()
+  const [article, setArticle] = useState(null);
```

Comment:
Ideally the value of state should be as consistent as possible, which in this case means this should be an object (an empty one by default) instead of null.